Fiks diverse ting `lintr` klagar på

haugedal opened the pull request:
Fiksar nokre av klagane frå `lintr`. Diverse ting som lange linjer, `1:nrow()`, `|` vs. `||`, osv.

Må ta denne etter at https://github.com/Rapporteket/rapwhale/pull/41 er fullført.

haugedal committed:
Bruk || og && i if()

haugedal committed:
Bruk seq_along() og seq_len() i staden for 1:length() og 1:nrow()

haugedal committed:
Bruk map_dbl() i staden for for-lykkje

haugedal committed:
Lagt til lintr-unntak

Brukar lange namn eller store bokstavar som kjem
frå andre pakkar.

haugedal committed:
Ny syntaks for anonyme funksjonar

haugedal committed:
Ikkje bruk store bokstavar i variabelnamn

haugedal committed:
Splitta lange linjer

haugedal committed:
Lagt til lintr-unntak for tilordning med <<-, := og variabelnamn med {}

Når det gjeld tilordning med <<- og := kan me kanskje unngå
nolint-kommentarar ved å få inn unntak for desse i
rapwhale:::assignment_eq_linter().

haugedal committed:
Fjerna unødvendig {} rundt kodeblokk

haugedal committed:
Fjerna utkommentert kode

haugedal committed:
Korrekt referanse til funksjon

Han stod i tillegg som vanleg kommentar (#),
og ikkje som roxygen-kommentar (#').

haugedal committed:
Lagt til nokre lintr-unntak

haugedal committed:
Bruk krøllparentesar for funksjonar lengre enn éi linje

Jf. kodestilguiden bør me vurdera om dei skal endrast:
Viss dei er lengre,
vurder sterkt å heller definera dei som vanlege,
namngjevne funksjonar.

haugedal committed:
Fjerna gamal dokumentasjon som fanst dobbelt opp

Same info ligg som roxygen-dokumentasjon.

haugedal committed:
Bruk seq_along() og seq_len() i staden for 1:length() og 1:nrow()

Her kan me òg unngå å bruka for-lykkjer,
men det ser me på i ei eiga grein.

haugedal commented:
@statkarl
No gjenstår berre nokre åtvaringar frå `cyclocomp_linter`. Nokre av desse kan me kanskje verta kvitt ved bruk av `checkmate`-pakken. Men me må kanskje gjera nett den testen mindre streng viss me skal setja `lintr`-sjekken som obligatorisk for PR-ar?

Elles kunne kanskje nokre av `lintr`-unntaka eg har lagt inn vore fiksa på andre måtar òg. Kom gjerne med innspel, eller fiks rett i greina.

statkarl commented on R/del-aar.R:
```diff
@@ -168,10 +168,8 @@ tid_til_tidslinje = function(dato, antall_deler) {
-  for (i in seq_along(aar_midtpunkt)) {
-    aar_midtpunkt[i] = (aar[i] + aar[i + 1]) / 2
-  }
+  aar_midtpunkt = seq_len(length(aar) - 1) |>
+    map_dbl(\(x) (aar[x] + aar[x + 1]) / 2)
```
Kan vel erstattast med

```r
aar_midtpunkt = (aar[-length(aar)] + aar[-1]) / 2
```

som er både raskare (vektorisert) og kanskje hakket lettare å forstå?

haugedal replied:
Gjort.

statkarl commented on R/utviklingsnivaa.R:
```diff
@@ -33,15 +33,15 @@ utviklingsnivaa = function(mappe = "man") {
-  parse_Rd_mapper = purrr::as_mapper(~ tools::parse_Rd(., permissive = TRUE))
+  parse_Rd_mapper = purrr::as_mapper(\(x) tools::parse_Rd(x, permissive = TRUE))
   funksjonsadresser = paste0(mappe, "/", funksjonar)
   funksjonar_parsed = map(funksjonsadresser, parse_Rd_mapper)
```
Bruk av `as_mapper()` er heilt overflødig her. Du gjer ein funksjon om til ein funksjon. La heller `parse_Rd_mapper()` vera ein heilt vanleg funksjon.

haugedal replied:
Fjerna `as_mapper()`.

statkarl replied:
Me brukar då ikkje `\(x)` utanom «inline». Endra til `function(x)`.

haugedal replied:
Endra.

statkarl commented on tests/testthat/test-les-kb-v2.R:
```diff
@@ -1,163 +1,163 @@
-  kb_eksempel = as_tibble(kb_eksempel)
-}
+)
+kb_eksempel = as_tibble(kb_eksempel)
+
```
Korfor vert objektet først laga som ein `data.frame()` (med `stringsAsFactors = FALSE`) for så etterpå bli gjort om til ein `tibble()`?

haugedal replied:
Heilt unødvendig. Gjort om.

statkarl commented on R/dformat-mrs.R:
```diff
@@ -362,7 +362,7 @@ les_dd_mrs = function(mappe_dd, skjema_id, versjon = "Prod", dato = NULL, kodebo
   #        opp for alle registera), gjer det mogleg å spesifisera ved
   #        funksjonskallet kva verdiar som skal tolkast til kva boolske verdiar.
-  #        Eks: boolske_verdiar = list(`FALSE` = "0", `TRUE` = "1", `NA` = c("", "-1", NA))
+  #        Eks: boolske_verdiar = list(`FALSE` = "0", `TRUE` = "1", `NA` = c("", "-1", NA)) # nolint
```
Denne var veldig uspesifikk. Kan du ikkje laga ein meir spesifikk regel (slik at det òg vert lettare å forstå kva som ikkje skal lintast)?

haugedal replied:
Det som ikkje skulle lintast var kommentert kode, men så måtte det verta lang linje i tillegg pga. linja vart lang med spesifikk nolint.

statkarl commented on R/dformat-oqr.R:
```diff
@@ -287,17 +287,19 @@ les_kb_oqr = function(mappe_dd, reg_id, dato = NULL, valider_kb = TRUE) { # fixm
       kandidatar = c(komb$namn[komb$id == kod_id[i]], kod_id[i])
       kandidatar = setdiff(kandidatar, kod_namn) # Fjern allereie brukte skjemanamn
-      kod_namn[i] = kandidatar[1] # Bruk første *ledige* (vil alltid vera ein, utanom det patologiske tilfelle der skjemanamna er lik skjema-ID-ane, men ikkje med 1-1-samsvar)
+      kod_namn[i] = kandidatar[1] # Bruk første *ledige* (vil alltid vera ein, utanom det patologiske tilfelle der skjemanamna er lik skjema-ID-ane, men ikkje med 1-1-samsvar) # nolint: line_length_linter.
```
Kommentaren kan vel heller skrivast (på fleire linjer) over kodelinja, så slepp me å nolinta.

haugedal replied:
Gjort.

statkarl commented on R/dformat-oqr.R:
```diff
@@ -73,7 +73,7 @@ les_kb_oqr = function(mappe_dd, reg_id, dato = NULL, valider_kb = TRUE) { # fixm
       listeverdier = col_character(),
       listetekst = col_character(),
-      normalintervall_start_numerisk = col_character(), # Sjå merknad nedanfor om årsaka til denne og dei tre neste må vera tekst
+      normalintervall_start_numerisk = col_character(), # Sjå merknad nedanfor om årsaka til denne og dei tre neste må vera tekst # nolint: line_length_linter.
```
Går vel an å skriva kommentaren på linja over og unngå nolinting.

haugedal replied:
Gjort.

statkarl commented on R/dformat-oqr.R:
```diff
@@ -287,17 +287,19 @@ les_kb_oqr = function(mappe_dd, reg_id, dato = NULL, valider_kb = TRUE) { # fixm
     # Bruk omkodingstabellen til gje ut rett namn på alle ID-ane
     kod_namn[match(ids, kod_id)]
   }
+  # nolint start commented_code_linter.
   # # Eksempel (og test)
   # ids   = c("pasreg",  "basereg", "basereg","pasreg",   "op",       "op",       "ev",       "basereg")
   # namn  = c("Pasient", "Basis",   "Basis",  "Opskjema", "Pasient",  "Opskjema", "Opskjema", "Basis")
   # fasit = c("Pasient", "Basis",   "Basis",  "Pasient",  "Opskjema", "Opskjema", "ev",       "Basis")
   # stopifnot(identical(fasit, tabell_id_til_skjemanamn(ids, namn)))
+  # nolint end
```
Dette bør heller gjerast til ein formell testthat()-testa (`tabell_id_til_skjemanamn()` må då takast ut som ein eigen funksjon).

haugedal replied:
Einig, men det kan vel vera eit eige Trello-kort og PR?

statkarl replied:
Så då lagar du eit slikt kort?

statkarl commented on R/dformat-oqr.R:
```diff
@@ -519,8 +521,10 @@ les_dd_oqr = function(mappe_dd, reg_id, skjema_id, status = 1, dato = NULL, kode
     str_detect(x, "^-?[0-9]+(\\.[0-9]+)?$")
   }
   ## Ev. kjapp test på at funksjonen fungerer som han skal
+  # nolint start: commented_code_linter.
   # stopifnot(all(er_tal(c("-3", "0", "1", "997", "3.14", "-3.14", "0.7"))))
   # stopifnot(all(!er_tal(c("a", "2B", "F42.7", "-x", "1e-7", "3.", ".7")))) # Ev. godta "3." og .7"?
+  # nolint end
```
Dette bør heller gjerast til ein formell testthat()-testa (`er_tal()` må då takast ut som ein eigen funksjon). Ev. finst det ein eigen talgjenkjenningsfunksjon i ein av pakkane me lastar, som me kan bruka?

haugedal replied:
Einig, men det kan vel vera eit eige Trello-kort og PR?

Fann ikkje ein slik funksjon i `checkmate`-pakken. Fann funksjonen `varhandle::check.numeric()` som ser ut til å gjera det me er ute etter? Men veit ikkje om me vil ha inn endå ein pakke berre for dette.

statkarl replied:
Nei, ikkje ny pakke inn. Du lagar Trello-kort?

statkarl commented on R/les_kb_v2.R:
```diff
@@ -418,7 +418,7 @@ tildel_unike_skjemanavn_fra_skjema_id = function(kb_std) {
     kandidatar = c(komb$namn[komb$id == kod_id[i]], kod_id[i])
     kandidatar = setdiff(kandidatar, kod_namn) # Fjern allereie brukte skjemanamn
-    kod_namn[i] = kandidatar[1] # Bruk første *ledige* (vil alltid vera ein, utanom det patologiske tilfelle der skjemanamna er lik skjema-ID-ane, men ikkje med 1-1-samsvar)
+    kod_namn[i] = kandidatar[1] # Bruk første *ledige* (vil alltid vera ein, utanom det patologiske tilfelle der skjemanamna er lik skjema-ID-ane, men ikkje med 1-1-samsvar) # nolint: line_length_linter.
```
Igjen kan kommentaren komma før kodelinja (og delt over fleire linjer).

haugedal replied:
Gjort.

statkarl commented on tests/testthat/test-datadump-valider.R:
```diff
@@ -70,6 +70,7 @@ test_that("Funksjonen stopper og rapporterer en feilmelding hvis kategoriske var
 
 #  Skal være mulig å ha bruke KB på ikke vårt standardformat, med å si at f.eks "min" heter "min_verdi" eller andre.
 
+# nolint start: commented_code_linter.
```
Korfor er alt dette kommentert ut?

haugedal replied:
Det veit eg ikkje, her er ikkje noko endra sidan langt tilbake til SVN-tida.

Men det ser ikkje ut som den utkommenterte testen testar det som var tenkt?

haugedal replied:
Eg lagar eit Trello kort på å rydda opp i dette, og så kan me få fullført denne PR-en?

statkarl replied:
OK.

statkarl commented on tests/testthat/test-les-kb-v2.R:
```diff
@@ -139,7 +139,7 @@ kb_eksempel = data.frame(
   lovlige_tegn = c(
     "1234567890", NA_character_,
-    "aAáÁbBcCdDðÐeEéÉfFgGhHiIíÍjJkKlLmMnNoOóÓpPqQrRsStTuUvúÚVwWxXyYýÝzZþÞæÆøØåÅäÄöÖñÑéÉ-ü ÜáÁ´_/,.- + 1234567890ØøÜüß@_-!%:;?",
+    "aAáÁbBcCdDðÐeEéÉfFgGhHiIíÍjJkKlLmMnNoOóÓpPqQrRsStTuUvúÚVwWxXyYýÝzZþÞæÆøØåÅäÄöÖñÑéÉ-ü ÜáÁ´_/,.- + 1234567890ØøÜüß@_-!%:;?", # nolint: line_length_linter.
```
Ev. bruka ein `paste0()`?

haugedal replied:
Gjort.

statkarl reviewed (changes requested):
Bra, men nokre småting som bør endrast.

statkarl commented:
> No gjenstår berre nokre åtvaringar frå `cyclocomp_linter`. Nokre av desse kan me kanskje verta kvitt ved bruk av `checkmate`-pakken. Men me må kanskje gjera nett den testen mindre streng viss me skal setja `lintr`-sjekken som obligatorisk for PR-ar?

Kanskje slå han heilt av? For (unødvendig) høg syklomatisk kompleksitet bør ein vel kunna oppdaga manuelt ved reviewprosessen.

haugedal committed:
Raskare (og enklare) kode

haugedal committed:
Fjerna overflødig bruk av as_mapper()

haugedal committed:
Lag eksempelkodebok direkte som tibble

haugedal committed:
Spesifikk nolinting

haugedal committed:
Flytta (og splitta) kommentarar for å unngå nolinting

haugedal committed:
Splitta linje med paste0() for å unngå nolint

haugedal committed:
Slått av cyclocomp_linter

Gjev mange uønska åtvaringar, spesielt for funksjonar som sjekkar
at inndata er korrekt. For (unødvendig) høg syklomatisk kompleksitet
bør ein vel kunna oppdaga manuelt ved reviewprosessen.

haugedal commented:
> > No gjenstår berre nokre åtvaringar frå `cyclocomp_linter`. Nokre av desse kan me kanskje verta kvitt ved bruk av `checkmate`-pakken. Men me må kanskje gjera nett den testen mindre streng viss me skal setja `lintr`-sjekken som obligatorisk for PR-ar?
> 
> Kanskje slå han heilt av? For (unødvendig) høg syklomatisk kompleksitet bør ein vel kunna oppdaga manuelt ved reviewprosessen.

Einig. Slo av sjekken.

Så no kan me kanskje setja `lintr`-sjekken som obligatorisk?

statkarl commented:
> Så no kan me kanskje setja `lintr`-sjekken som obligatorisk?

Ja. Kan alltids slå han av igjen viss det viser seg å vera veldig problematisk.

haugedal committed:
Ikkje bruk \(x) utanom «inline»
